### PR TITLE
Simplify silence modifications, add update endpoint

### PR DESCRIPTION
**Pull request by fabxc**
@mxinden

### CODE REVIEW

**Comment by mxinden** [comment]
Fixes issue #765

File: silence/silence.go [hunk 1]
```diff
@@ -366,115 +373,87 @@ func (s *Silences) setSilence(sil *pb.Silence) error {
...
-	s.mtx.Lock()
-	defer s.mtx.Unlock()
+	return sil.Id, s.setSilence(sil)
+}
 
-	if err := s.setSilence(sil); err != nil {
-		return "", err
+// canUpdate returns true if silence a can be updated to b without
+// affecting the historic view of silencing.
+func canUpate(a, b *pb.Silence, now time.Time) bool {
```
Comment (stuartnelson3) [inline silence/silence.go]: `/s/canUpate/canUpdate/`

File: silence/silence.go [hunk 2]
```diff
@@ -562,6 +541,19 @@ func QState(states ...SilenceState) QueryParam {
 	}
 }
 
+// QueryOne queries with the given paramters and returns the first result.
```
Comment (stuartnelson3) [inline silence/silence.go]: `/s/paramters/parameters/`

File: ui/app/src/Silences/Api.elm [hunk 3]
```diff
@@ -37,7 +37,7 @@ create silence =
             String.join "/" [ baseUrl, "silences" ]
 
         body =
-            Http.jsonBody <| Silences.Encoders.silence silence
+            Http.jsonBody <| Silences.Encoders.silence <| Debug.log "bla" silence
```
Comment (mxinden) [inline ui/app/src/Silences/Api.elm]: @fabxc Can you remove the Debug?

**Comment by fabxc** [comment]
Updated

**Comment by mxinden** [comment]
@fabxc `bindata.go` still seems to be outdated (See failing CI).

**Comment by fabxc** [comment]
Done

**Comment by mxinden** [comment]
@stuartnelson3 Any other comments? Otherwise I will merge here.

**Pull request merged**